mitmproxy/flow.py

```diff
@@ -82,9 +82,18 @@ class Flow(stateobject.StateObject):
     We're waiting for a user action to forward the flow to its destination.
     """
 
-    marked: bool
+    marked: typing.Union[bool, str]
```

Comment:
Let's make this `str`. `""` denotes "no marker" and implicitly evaluates to `False`, so we can do keep saying `if flow.marker: ...`. Having this as a single type makes it much easier to work with than having to check if it's a bool or a str all the time.

Reply:
I've made the `marker` attribute a `str` type, only, everywhere. This propagated a bit into the test code, too.

I added a `:default:` marker that provides `SYMBOL_MARK` as a convenience for those who want the exact same marker as before.

The `Marker` type will still accept `true` and `false` for when the user issues commands or for use in keystroke bindings. This just seemed a useful accommodation.